openwisp_controller/connection/models.py

```diff
@@ -225,3 +230,14 @@ def update_config(self):
             else:
                 self.device.config.set_status_applied()
                 self.disconnect()
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        if self.is_working != self._initial_status:
+            self.send_working_status_changed_signal()
+            self.last_attempt = now()
+        self._initial_status = self.is_working
+
+    def send_working_status_changed_signal(self):
+        working_status_changed = Signal(providing_args=['working_status'])
```

Comment:
define this in a dedicated file, like here: https://github.com/openwisp/openwisp-monitoring/blob/master/openwisp_monitoring/monitoring/signals.py